src/api/goods.ts

```
@@ -8,9 +8,13 @@ export function getAll(): Promise<Good[]> {
 }
 
 export const get5First = () => {
-  return getAll().then(goods => goods); // sort and get the first 5
+  return getAll().then(goods =>
+    goods
+      .sort((good1, good2) => good1.name.localeCompare(good2.name))
+      .slice(0, 5),
+  ); // sort and get the first 5
```

This function, along with `getRedGoods`, relies on `getAll()`. However, the current implementation of `getAll()` is missing proper error handling.

This violates checklist item #3: 'When you make a request to the server, don't forget to handle possible errors'.

The `fetch` promise only rejects on network errors, not on HTTP error statuses like 404 or 500. To fix this, you should modify the `getAll` function to check the `response.ok` property. If it's `false`, you should throw an error to make the promise reject, allowing the `catch` block in `App.tsx` to handle it correctly.